stm/stm.mli

```diff
@@ -182,14 +189,9 @@ val register_proof_block_delimiter :
  * the name of the Task(s) above) *)
 
 val state_computed_hook : (Stateid.t -> in_cache:bool -> unit) Hook.t
-val parse_error_hook :
-  (Feedback.edit_or_state_id -> Loc.t -> Pp.std_ppcmds -> unit) Hook.t
-val unreachable_state_hook : (Stateid.t -> Exninfo.iexn -> unit) Hook.t
+
 (* ready means that master has it at hand *)
 val state_ready_hook : (Stateid.t -> unit) Hook.t
-(* called with true before and with false after a tactic explicitly
- * in the document is run *)
-val tactic_being_run_hook : (bool -> unit) Hook.t
```

Comment:
This PR is really about cruft removal so this is why that was included there. IMHO we have accumulated too much cruft and duplication in the codebase, 8.7 could be a good opportunity to clear things ups a bit.

That being said, just let me know what you think and I'll drop that patch.